base/dict.jl

```diff
@@ -89,67 +146,137 @@ Dict{String,Int64} with 2 entries:
   "A" => 1
 ```
 """
-mutable struct Dict{K,V} <: AbstractDict{K,V}
-    slots::Array{UInt8,1}
-    keys::Array{K,1}
-    vals::Array{V,1}
-    ndel::Int
-    count::Int
-    age::UInt
-    idxfloor::Int  # an index <= the indexes of all used slots
-    maxprobe::Int
+mutable struct Dict{K,V} <: HashDict{K,V}
```

Comment:
not mutable. also, we should optimize to remove the allocation for this case (since it would be relatively trivial), but currently don't, so we may need to make sure the perf regression is minor in the meantime.

Reply:
Yes, not mutable, thanks.  And I was not aware of the perf problem.  I'll run it against https://github.com/JuliaCI/BaseBenchmarks.jl/pull/150 and some against older perf-tests.